File: 🟡 Medium/53 - Maximum Subarray/_53MaximumSubarray.cpp
class Solution {
    public:
        int maxSubArray(vector<int>& nums) {
            // KADANE'S ALGO
    
            int cs = 0;
            int maxSum = INT_MIN;
    
            for (auto& num : nums) {
                cs += num;
                maxSum = max(maxSum, cs);
    
                if (cs < 0) {
                    cs = 0;
                }
            }
    
            return maxSum;
        }
    };

File: 🟡 Medium/53 - Maximum Subarray/_53MaximumSubarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
    // KADANE'S ALGORITHM
    let cs = 0;
    let maxSum = -Infinity;

    for (let num of nums) {
        cs += num;
        maxSum = Math.max(maxSum, cs);

        if (cs < 0) {
            cs = 0;
        }
    }

    return maxSum;
};